### PR TITLE
openshot-qt: 2.2.0 -> 2.3.1

### PR DESCRIPTION
**Pull request by therealpxc**
###### Motivation for this change
New upstream release (also trying to convince Bryan Lunduke to talk about Nix :-P)

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

When I ran this on my system (NixOS 17.03), it worked fine there, but when I tried to run the version I successfully built against the Nixpkgs master branch, I got an error about how Qt couldn't find the XCB plugin. I'm not sure if this would be an issue if I was running on NixOS unstable channel or master.

### CODE REVIEW

**Comment by Mic92** [comment]
I simplified the wrapper code a little bit, but loading the xcb plugins remains an error on master.
When applying `export QT_DEBUG_PLUGINS=1` in my shell, I get the following error:

https://gist.github.com/Mic92/7c73214090560627222040abc9a48545

File: pkgs/applications/video/openshot-qt/default.nix [hunk 1]
```diff
@@ -23,12 +25,9 @@ stdenv.mkDerivation rec {
   installPhase = ''
```
Comment (ttuegel) [inline pkgs/applications/video/openshot-qt/default.nix]: If you override `installPhase`, the first line must be `runHook preInstall` and the last line must be `runHook postInstall`.

File: pkgs/applications/video/openshot-qt/default.nix [hunk 2]
```diff
@@ -23,12 +25,9 @@ stdenv.mkDerivation rec {
...
     mkdir -p $(toPythonPath $out)
     cp -r src/* $(toPythonPath $out)
-    mkdir -p $out/bin
-    echo "#/usr/bin/env sh" >$out/bin/openshot-qt
-    echo "exec ${python3Packages.python.interpreter} $(toPythonPath $out)/launch.py" >>$out/bin/openshot-qt
-    chmod +x $out/bin/openshot-qt
-    wrapQtProgram $out/bin/openshot-qt \
-      --prefix PYTHONPATH : "$(toPythonPath $out):$(toPythonPath ${libopenshot}):$(toPythonPath ${python3Packages.pyqt5}):$(toPythonPath ${python3Packages.sip}):$(toPythonPath ${python3Packages.httplib2}):$(toPythonPath ${python3Packages.pyzmq}):$PYTHONPATH"
+
+    makeQtWrapper $(toPythonPath $out)/launch.py $out/bin/openshot-qt \
```
Comment (ttuegel) [inline pkgs/applications/video/openshot-qt/default.nix]: It is not a good idea to call `makeQtWrapper` until `preFixup`.

**Comment by ttuegel** [comment]
> looks like version mismatch, but i cannot tell why.

It looks to me like a linker error rather than a version mismatch. (Version mismatches always result in a segmentation fault.) It may be related to the issues I commented on.

**Comment by Mic92** [comment]
Looks like using `buildPythonApplication` works fine. Can you confirm that video editing still works?

**Comment by Mic92** [comment]
Open dialogs are now working in my environment.

**Comment by bjornfor** [comment]
Should this be picked to NixOS 17.03?

**Comment by Mic92** [comment]
@bjornfor https://github.com/NixOS/nixpkgs/pull/24868/commits/deeddbd495dffe78163430aa49b2487d9d10681c is probably interesting for Gnome user. I have no opinion about the changes.

**Pull request merged**